packages/react-query/src/utils/structuralSharing.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * Taken from @tanstack/query-core utils.ts
+ * Modified to support Date, Map, Set and NaN comparisons
+ *
+ * This function returns `a` if `b` is deeply equal.
+ * If not, it will replace any deeply equal children of `b` with those of `a`.
+ */
+export function replaceEqualDeep(a: any, b: any): any {
```

Comment:
Since we have this as the only dependency instead, can we just import it instead?

Reply:
As discussed, I updated the wording.